frontend/src/libs/helpers/test-server-error-type/test-server-error-type.helper.ts

```diff
@@ -0,0 +1,38 @@
+import { ServerErrorType } from '~/libs/enums/enums.js';
+import {
+  type ServerCommonErrorResponse,
+  type ServerValidationErrorResponse,
+} from '~/libs/types/types.js';
+
+import { objectPick } from '../helpers.js';
+
+// This is to reassure this key name at the typescript level
+const errorTypeKey: keyof (
+  | ServerCommonErrorResponse
+  | ServerValidationErrorResponse
+) = 'errorType';
+
+const testServerErrorType = (
+  error: unknown,
+): ServerCommonErrorResponse | ServerValidationErrorResponse | undefined => {
+  if (!!error && typeof error === 'object' && errorTypeKey in error) {
+    if (error.errorType === ServerErrorType.COMMON) {
+      return objectPick(error as ServerCommonErrorResponse, [
+        'errorType',
+        'message',
+      ]);
```

Comment:
not sure why we need pick helper, it can be used only on top-level keys. We can just use destructuring 
```const { errorType, message } = error

return {
errorType, message
}
```
```
import pick from 'object.pick';
import { pick as _pick} from 'lodash';

const foo = {
  a: [{ b: 2}, { c: 3 }]
}

console.log(pick(foo, [
  'a.0'
])) 
// {}

console.log(_pick(foo, [
  'a.0'
]))
//  { a: [{ b: 2}] }
```